doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+      - [Manifest](#manifest)
+      - [CLI/YANG Model Enhancements](#cliyang-model-enhancements)
+        - [SNOiC VRRP YANG MODEL](#sonic-vrrp-yang-model)
+		- [CLI](#cli)
+          - [Config commands](#config-commands)
+          - [Show commands](#show-commands)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+    - [Open/Action items](#openaction-items)
+ 
+### Revision  
+|  Rev  |  Date           |  Author     | Change Description |
+| :---  | :-------------- | :---------- | :----------------  |
+|  0.1  |  Aug-16-2023    | Philo-micas | Initial version    |
+
+### Scope  
+
+This document describes the high level design of frr-vrrpd adaptation to SONiC.
+
+### Definitions/Abbreviations 
+
+Table 1: Abbreviations
+
+| Abbreviation    | Description                                        |
+| :----------     | :------------------------------------------------  |
+| VRRP            | Virtual Router Redundency Protocol                 |
+| ARP             | Address Resolution Protocol                        |
+| FRR             | Free Range Routing Stack                           |
+| CLI             | Command Line Interface                             |
+| VMAC            | Virtual MAC address                                |
+| VIP             | Virtual IP address                                 |
+| VRID            | Virtual Router Identifier                          |
+| VRRP Instance   | An instance of VRRP state machine on an interface. Multiple VRRP state machines can be configured on an interface.         |
+| VRRP Owner      | VRRP owner (of a VRRP instance) is the router whose virtual IP address is the same as the real interface IP address        |
+
+### Overview 
+This document describes the high level design details of VRRP and its components. VRRP(Virtual Router Redundancy Protocol) specifies an election protocol that dynamically assigns responsibility for a virtual router to one of the VRRP routers on a LAN.
+The election provides two basic functionality:
+	support dynamic failover in the forwarding responsibility when the Master become unavailable.
+	support load sharing among multiple routers by running multiple instances.
+
+
+### Requirements
+
+#### Functional Requirements
+
+Following requirements are addressed by the design presented in this document:
+  1. Support VRRPv2(IPv4) and VRRPv3(IPv4 and IPv6)
+  2. Support VRRP on Ethernet, VLAN and PortChannel interfaces
+  3. Support interface configuration of multiple VRRP instances.
+  4. Support configurable priority for VRRP instance
+  5. Support configurable preempt mode for VRRP instance
+  6. Support uplink interface tracking feature
+
```

Comment:
Is sub-interface, VRF and owner case supported?

Reply:
Currently, the sub interface is waiting for validation; and VRF is supported;
due to causing lots of different issues, FRR has temporarily disabled the owner function.